tests/ode_test.cpp

```diff
@@ -79,3 +87,24 @@ TEST(OdeTest, Ode4ShoTest) {
   ASSERT_FLOAT_EQ(y[3][1], -0.295519962530663);
   // ASSERT_FLOAT_EQ(y[3][1], -0.295520962530663); Fails as expected
 }
+
+TEST(OdeTest, Ode23ShoTest) {
+  PAN_GNC_ODEXX_TEST_VARS(6);
+  int code;
+  // Check against cos(1.0) and assert the error code is OK
+  code = gnc::ode23(ti, 1.0, yi, yf, ne, bf, 1e-4, 1e-6, 1e-6, 1000, fsho);
+  ASSERT_EQ(code, gnc::ODE_ERR_OK);
+  ASSERT_NEAR(yf[0], cos(1.0), 1e-3);
+  // Check against cos(5.0) and assert the error code is OK
+  code = gnc::ode23(ti, 5.0, yi, yf, ne, bf, 1e-4, 1e-6, 1e-6, 1000, fsho);
+  ASSERT_EQ(code, gnc::ODE_ERR_OK);
+  ASSERT_NEAR(yf[0], cos(5.0), 1e-3);
+  // Check against cos(6.5) and assert the error code is OK
+  code = gnc::ode23(ti, 6.5, yi, yf, ne, bf, 1e-4, 1e-6, 1e-6, 1000, fsho);
+  ASSERT_EQ(code, gnc::ODE_ERR_OK);
+  ASSERT_NEAR(yf[0], cos(6.5), 1e-3);
+  // Check against cos(7.5) and assert the error code is OK
+  code = gnc::ode23(ti, 7.5, yi, yf, ne, bf, 1e-4, 1e-6, 1e-6, 2000, fsho);
+  ASSERT_EQ(code, gnc::ODE_ERR_OK);
+  ASSERT_NEAR(yf[0], cos(7.5), 2e-3);
```

Comment:
The required tolerance here was calculated as `top * max_iters` (i.e. worts case miss on up to the maximum number of iterations). In practice, of course, this generally doesn't happen. For the sake of the unit test, however, that's how I picked the assertion tolerance.